Primo.js

```diff
@@ -213,7 +213,13 @@ function importPNX(text, url) {
 	
 	var itemType = ZU.xpathText(doc, '//display/type');
 	if(!itemType) {
-		throw new Error('Could not locate item type');
+		if ( ZU.xpathText(doc, '//facets/rsrctype') ) {
+			var itemType = ZU.xpathText(doc, '//facets/rsrctype');
+		} else if ( ZU.xpathText(doc, '//search/rsrctype') ) {
+			var itemType = ZU.xpathText(doc, '//search/rsrctype');
```

Comment:
There is also a type field in the `addata` section and especially there is a `addata/ristype`, which should be more standardized. Did you try this out? Actually, I would even try to use it as first option.